python/ray/data/grouped_data.py

```diff
@@ -107,6 +107,22 @@ class PushBasedGroupbyOp(_GroupbyOp, PushBasedShufflePlan):
     pass
 
 
+class _MultiColumnSortedKey:
+    """Acting as a tuple of group keys for np.searchsorted"""
```

Comment:
Could you add an implementation note describing why we need to define this class?